diff -ur qtwebengine-opensource-src-5.6.0-beta/src/core/config/linux.pri qtwebengine-opensource-src-5.6.0-beta-linux-pri/src/core/config/linux.pri
--- qtwebengine-opensource-src-5.6.0-beta/src/core/config/linux.pri	2015-12-14 16:27:24.000000000 +0100
+++ qtwebengine-opensource-src-5.6.0-beta-linux-pri/src/core/config/linux.pri	2016-01-06 04:43:23.115923478 +0100
@@ -43,3 +43,20 @@
 use?(system_icu):      GYP_CONFIG += use_system_icu=1
 use?(system_ffmpeg):   GYP_CONFIG += use_system_ffmpeg=1
 
+# some more system packages, that are not checked for for some reason
+GYP_CONFIG += \
+    use_system_yasm=1 \
+    use_system_expat=1 \
+    use_system_re2=1 \
+    use_system_protobuf=1
+
+# link libpci instead of dlopening it, our Qt packaging depends on it anyway
+GYP_CONFIG += linux_link_libpci=1
+
+# run the unbundling script Chromium provides
+CHROMIUM_SRC_DIR = "$$QTWEBENGINE_ROOT/$$getChromiumSrcDir()"
+R_G_F_PY = "$$CHROMIUM_SRC_DIR/build/linux/unbundle/replace_gyp_files.py"
+R_G_F_PY_ARGS =
+for (config, GYP_CONFIG): R_G_F_PY_ARGS += "-D $$config"
+log("Running python $$R_G_F_PY $$R_G_F_PY_ARGS$${EOL}")
+!system("python $$R_G_F_PY $$R_G_F_PY_ARGS"): error("-- unbundling failed")
