diff -ur qtwebengine-opensource-src-5.7.1/src/3rdparty/chromium/third_party/webrtc/common_audio/common_audio.gyp qtwebengine-opensource-src-5.7.1-webrtc-neon/src/3rdparty/chromium/third_party/webrtc/common_audio/common_audio.gyp
--- qtwebengine-opensource-src-5.7.1/src/3rdparty/chromium/third_party/webrtc/common_audio/common_audio.gyp	2016-11-07 15:46:18.000000000 +0100
+++ qtwebengine-opensource-src-5.7.1-webrtc-neon/src/3rdparty/chromium/third_party/webrtc/common_audio/common_audio.gyp	2016-12-04 01:07:09.731128900 +0100
@@ -217,7 +217,31 @@
         {
           'target_name': 'common_audio_neon',
           'type': 'static_library',
-          'includes': ['../build/arm_neon.gypi',],
+          'cflags!': [
+            '-mfpu=vfpv3-d16',
+          ],
+          'cflags_c!': [
+            '-mfpu=vfpv3-d16',
+          ],
+          'cflags_cc!': [
+            '-mfpu=vfpv3-d16',
+          ],
+          'conditions': [
+            # "-mfpu=neon" is not required for arm64 in GCC.
+            ['target_arch!="arm64"', {
+              'cflags': [
+                '-mfpu=neon',
+              ],
+            }],
+            # Disable GCC LTO on NEON targets due to compiler bug.
+            # TODO(fdegans): Enable this. See crbug.com/408997.
+            ['clang==0 and use_lto==1', {
+              'cflags!': [
+                '-flto',
+                '-ffat-lto-objects',
+              ],
+            }],
+          ],
           'sources': [
             'fir_filter_neon.cc',
             'resampler/sinc_resampler_neon.cc',
diff -ur qtwebengine-opensource-src-5.7.1/src/3rdparty/chromium/third_party/webrtc/modules/audio_coding/codecs/isac/isacfix.gypi qtwebengine-opensource-src-5.7.1-webrtc-neon/src/3rdparty/chromium/third_party/webrtc/modules/audio_coding/codecs/isac/isacfix.gypi
--- qtwebengine-opensource-src-5.7.1/src/3rdparty/chromium/third_party/webrtc/modules/audio_coding/codecs/isac/isacfix.gypi	2016-11-07 15:46:18.000000000 +0100
+++ qtwebengine-opensource-src-5.7.1-webrtc-neon/src/3rdparty/chromium/third_party/webrtc/modules/audio_coding/codecs/isac/isacfix.gypi	2016-12-04 01:07:02.813235612 +0100
@@ -128,7 +128,31 @@
         {
           'target_name': 'isac_neon',
           'type': 'static_library',
-          'includes': ['../../../../build/arm_neon.gypi',],
+          'cflags!': [
+            '-mfpu=vfpv3-d16',
+          ],
+          'cflags_c!': [
+            '-mfpu=vfpv3-d16',
+          ],
+          'cflags_cc!': [
+            '-mfpu=vfpv3-d16',
+          ],
+          'conditions': [
+            # "-mfpu=neon" is not required for arm64 in GCC.
+            ['target_arch!="arm64"', {
+              'cflags': [
+                '-mfpu=neon',
+              ],
+            }],
+            # Disable GCC LTO on NEON targets due to compiler bug.
+            # TODO(fdegans): Enable this. See crbug.com/408997.
+            ['clang==0 and use_lto==1', {
+              'cflags!': [
+                '-flto',
+                '-ffat-lto-objects',
+              ],
+            }],
+          ],
           'dependencies': [
             '<(webrtc_root)/common_audio/common_audio.gyp:common_audio',
           ],
diff -ur qtwebengine-opensource-src-5.7.1/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/audio_processing.gypi qtwebengine-opensource-src-5.7.1-webrtc-neon/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/audio_processing.gypi
--- qtwebengine-opensource-src-5.7.1/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/audio_processing.gypi	2016-11-07 15:46:18.000000000 +0100
+++ qtwebengine-opensource-src-5.7.1-webrtc-neon/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/audio_processing.gypi	2016-12-04 01:06:58.455302835 +0100
@@ -260,7 +260,31 @@
       'targets': [{
         'target_name': 'audio_processing_neon',
         'type': 'static_library',
-        'includes': ['../../build/arm_neon.gypi',],
+        'cflags!': [
+          '-mfpu=vfpv3-d16',
+        ],
+        'cflags_c!': [
+          '-mfpu=vfpv3-d16',
+        ],
+        'cflags_cc!': [
+          '-mfpu=vfpv3-d16',
+        ],
+        'conditions': [
+          # "-mfpu=neon" is not required for arm64 in GCC.
+          ['target_arch!="arm64"', {
+            'cflags': [
+              '-mfpu=neon',
+            ],
+          }],
+          # Disable GCC LTO on NEON targets due to compiler bug.
+          # TODO(fdegans): Enable this. See crbug.com/408997.
+          ['clang==0 and use_lto==1', {
+            'cflags!': [
+              '-flto',
+              '-ffat-lto-objects',
+            ],
+          }],
+        ],
         'dependencies': [
           '<(webrtc_root)/common_audio/common_audio.gyp:common_audio',
         ],
diff -ur qtwebengine-opensource-src-5.7.1/src/3rdparty/chromium/third_party/webrtc/modules/video_processing/video_processing.gypi qtwebengine-opensource-src-5.7.1-webrtc-neon/src/3rdparty/chromium/third_party/webrtc/modules/video_processing/video_processing.gypi
--- qtwebengine-opensource-src-5.7.1/src/3rdparty/chromium/third_party/webrtc/modules/video_processing/video_processing.gypi	2016-11-07 15:46:18.000000000 +0100
+++ qtwebengine-opensource-src-5.7.1-webrtc-neon/src/3rdparty/chromium/third_party/webrtc/modules/video_processing/video_processing.gypi	2016-12-04 01:06:53.268382845 +0100
@@ -47,7 +47,7 @@
         ['target_arch=="ia32" or target_arch=="x64"', {
           'dependencies': [ 'video_processing_sse2', ],
         }],
-        ['target_arch=="arm" or target_arch == "arm64"', {
+        ['build_with_neon==1', {
           'dependencies': [ 'video_processing_neon', ],
         }],
       ],
@@ -77,12 +77,36 @@
         },
       ],
     }],
-    ['target_arch=="arm" or target_arch == "arm64"', {
+    ['build_with_neon==1', {
       'targets': [
         {
           'target_name': 'video_processing_neon',
           'type': 'static_library',
-          'includes': [ '../../build/arm_neon.gypi', ],
+          'cflags!': [
+            '-mfpu=vfpv3-d16',
+          ],
+          'cflags_c!': [
+            '-mfpu=vfpv3-d16',
+          ],
+          'cflags_cc!': [
+            '-mfpu=vfpv3-d16',
+          ],
+          'conditions': [
+            # "-mfpu=neon" is not required for arm64 in GCC.
+            ['target_arch!="arm64"', {
+              'cflags': [
+                '-mfpu=neon',
+              ],
+            }],
+            # Disable GCC LTO on NEON targets due to compiler bug.
+            # TODO(fdegans): Enable this. See crbug.com/408997.
+            ['clang==0 and use_lto==1', {
+              'cflags!': [
+                '-flto',
+                '-ffat-lto-objects',
+              ],
+            }],
+          ],
           'sources': [
             'util/denoiser_filter_neon.cc',
             'util/denoiser_filter_neon.h',
